server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/UnifiedHighlighter.java

```diff
@@ -145,7 +142,27 @@ public HighlightField highlight(HighlighterContext highlighterContext) {
         return null;
     }
 
-    private BreakIterator getBreakIterator(SearchContextHighlight.Field field) {
+    protected PassageFormatter getPassageFormatter(SearchContextHighlight.Field field, Encoder encoder) {
+        CustomPassageFormatter passageFormatter = new CustomPassageFormatter(field.fieldOptions().preTags()[0],
+            field.fieldOptions().postTags()[0], encoder);
+        return passageFormatter;
+    }
+
+    
+    protected Analyzer wrap(Analyzer analyzer) {
+        return analyzer;
+    }
+    
```

Comment:
Can we merge wrap() into getAnalyzer()?  It feels wrong to have both methods

Reply:
getAnalyzer is static so can't be overriden?

Reply:
Static method prob needs to stay as already called from PlainHighlighter and can't have instance method with same name so least disruptive change is a new `getThisAnalyzer()` (not crazy about that name) that calls existing static getAnalyzer.

Reply:
How about moving the static method to `HighlightUtils`?  That seems to me to be a better place for it anyway, if it's being called from multiple highlighters.